File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/service/HistoryService.kt
package com.windea.demo.cloudcollect.core.service

import com.windea.demo.cloudcollect.core.domain.entity.*
import org.springframework.data.domain.*

interface HistoryService {
	/**创建浏览记录。*/
	fun create(history: History, user: User): History
	
	/**删除一条浏览记录。*/
	fun deleteById(id: Long)
	
	/**删除某一用户的所有浏览记录。*/
	fun deleteAllByUserId(userId: Long)
	
	/**得到某一用户的所有浏览记录。*/
	fun findAllByUserId(userId: Long, pageable: Pageable): Page<History>
}

File: cloud-collect-web/src/service/historyService.ts
import http from "@/http"
import {History, Page, PageableParam} from "@/types"

const historyUrl = "/history"

export async function create(history: History) {
  return await http.post(`${historyUrl}/create`, history)
}

export async function deleteById(id: number) {
  return await http.delete(`${historyUrl}/${id}`)
}

export async function deleteAllByUserId(userId: number) {
  const params = {userId}
  return await http.delete(`${historyUrl}/deleteAllByUserId`, {params})
}

export async function findAllByUserId(userId: number, pageableParam: PageableParam) {
  const params = {userId, ...pageableParam}
  return await http.get<Page<History>>(`${historyUrl}/findAllByUserId`, {params})
}
